### DISCUSSION TITLE
make[1]: *** No rule to make target 'clean'.  Stop.

### DISCUSSION THREAD

**Discussion by blockchainx** [filecoin-project/lotus#8880] [Build and Sync] [answered]
Ubuntu 20.04.1

Action:
git clone https://github.com/filecoin-project/lotus.git
cd lotus/
make clean all

Actual result:
root@ubuntu:~/lotus# make clean all
rm -rf  build/.filecoin-install build/.update-modules  lotus lotus-miner lotus-worker lotus-shed lotus-gateway lotus-seed lotus-pond lotus-fountain lotus-bench lotus-stats lotus-pcr lotus-health lotus-wallet lotus-keygen testground tvx lotus-sim
make -C extern/filecoin-ffi/ clean
make[1]: Entering directory '/root/lotus/extern/filecoin-ffi'
make[1]: *** No rule to make target 'clean'.  Stop.
make[1]: Leaving directory '/root/lotus/extern/filecoin-ffi'
make: [Makefile:289: clean] Error 2 (ignored)
git submodule update --init --recursive
Cloning into '/root/lotus/extern/test-vectors'...
error: RPC failed; curl 56 GnuTLS recv error (-9): Error decoding the received TLS packet.
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
fatal: clone of 'https://github.com/filecoin-project/test-vectors.git' into submodule path '/root/lotus/extern/test-vectors' failed
Failed to clone 'extern/test-vectors'. Retry scheduled
Cloning into '/root/lotus/extern/test-vectors'...

**Comment by LexLuthr** [comment, marked as answer]
This is a normal behavior and you can ignore the errors.

Why it happens?
1. Make runs "clean" before "all"
2. At this point no submodule is initialised. 
3. So, when make run "make -C extern/filecoin-ffi clean" , there is no file inside "extern/filecoin-ffi"
4. This results in the error you see.

If you want to test this theory then run the following steps:

1. git clone https://github.com/filecoin-project/lotus.git
2. cd lotus/
3. git submodule update --init
4. make clean all

You should not see any errors as submodules have been downloaded and Makefile for filecoin-ffi exists on the system.

**Comment by blockchainx** [comment]
run "git submodule update --init"
----shows "GnuTLS recv error (-54): Error in the pull function.", reboot the system and try again, it works.

run "make clean all"
----it stuck in downloading "go: downloading github.com/ipfs/go-ipfs-blockstore v1.1.2", see log below:

root@ubuntu:~/lotus# make clean all
rm -rf build/.filecoin-install build/.update-modules lotus lotus-miner lotus-worker lotus-shed lotus-gateway lotus-seed lotus-pond lotus-fountain lotus-bench lotus-stats lotus-pcr lotus-health lotus-wallet lotus-keygen testground tvx lotus-sim
make -C extern/filecoin-ffi/ clean
make[1]: Entering directory '/root/lotus/extern/filecoin-ffi'
go clean -cache -testcache .
go: downloading github.com/filecoin-project/go-address v0.0.6
go: downloading github.com/filecoin-project/go-fil-commcid v0.1.0
go: downloading github.com/filecoin-project/go-state-types v0.1.8
go: downloading github.com/ipfs/go-cid v0.1.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading github.com/ipfs/go-block-format v0.0.3
go: downloading github.com/ipfs/go-ipfs-blockstore v1.1.2
rm -rf filcrypto.h filcrypto.pc libfilcrypto.a .install-filcrypto
rm -f ./runner
cd rust && cargo clean && cd ..
make[1]: Leaving directory '/root/lotus/extern/filecoin-ffi'
git submodule update --init --recursive
touch build/.update-modules
make -C extern/filecoin-ffi/ .install-filcrypto
make[1]: Entering directory '/root/lotus/extern/filecoin-ffi'
go clean -cache -testcache .
go: downloading github.com/filecoin-project/go-address v0.0.6
go: downloading github.com/filecoin-project/go-fil-commcid v0.1.0
go: downloading github.com/filecoin-project/go-state-types v0.1.8
go: downloading github.com/ipfs/go-cid v0.1.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading github.com/ipfs/go-block-format v0.0.3
go: downloading github.com/ipfs/go-ipfs-blockstore v1.1.2

**Comment by LexLuthr** [comment]
```
run "git submodule update --init"
----shows "GnuTLS recv error (-54): Error in the pull function.", reboot the system and try again, it works.
```

This seems like a system command issue. You need to check this on your side. Nothing to do with lotus.

Stuck download could be a network issue or issue with golang version. This is again not specific to lotus.

**Reply by blockchainx** [reply]
may be wait for tens of minutes, now it continues downloading, it shall works now, many thanks.

**Reply by blockchainx** [reply]
add the commands to speed up.
export IPFS_GATEWAY=https://proof-parameters.s3.cn-south-1.jdcloud-oss.com/ipfs/
export GOPROXY=https://goproxy.cn
